osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetPowerQualityProfileCommandExecutor.java

```diff
@@ -305,33 +399,50 @@ private String getUnit(final ScalerUnitInfo scalerUnitInfo) {
     }
 
     private List<ProfileEntryValueDto> makeProfileEntryValueDto(final DataObject profEntryDataObjects,
-            final List<ScalerUnitInfo> scalerUnitInfos) throws ProtocolAdapterException {
+            final List<ScalerUnitInfo> scalerUnitInfos, ProfileEntryDto previousProfileEntryDto, int timeInterval) {
 
         final List<ProfileEntryValueDto> result = new ArrayList<>();
+
         final List<DataObject> dataObjects = profEntryDataObjects.getValue();
         for (int i = 0; i < dataObjects.size(); i++) {
-            result.add(this.makeProfileEntryValueDto(dataObjects.get(i), scalerUnitInfos.get(i)));
+
+            ProfileEntryValueDto currenProfileEntryValueDto = this
+                    .makeProfileEntryValueDto(dataObjects.get(i), scalerUnitInfos.get(i), previousProfileEntryDto,
+                            timeInterval);
+            result.add(currenProfileEntryValueDto);
+
         }
         return result;
     }
 
     private ProfileEntryValueDto makeProfileEntryValueDto(final DataObject dataObject,
-            final ScalerUnitInfo scalerUnitInfo) throws ProtocolAdapterException {
+            final ScalerUnitInfo scalerUnitInfo, ProfileEntryDto previousProfileEntryDto, int timeInterval) {
         if (InterfaceClass.CLOCK.id() == scalerUnitInfo.getClassId()) {
-            return this.makeDateProfileEntryValueDto(dataObject);
+            return this.makeDateProfileEntryValueDto(dataObject, previousProfileEntryDto, timeInterval);
         } else if (dataObject.isNumber()) {
             return this.makeNumericProfileEntryValueDto(dataObject, scalerUnitInfo);
         } else {
             final String dbgInfo = this.dlmsHelper.getDebugInfo(dataObject);
-            LOGGER.debug("creating ProfileEntryDto from " + dbgInfo + " " + scalerUnitInfo);
+            LOGGER.debug("creating ProfileEntryDto from {} {} ", dbgInfo, scalerUnitInfo);
             return new ProfileEntryValueDto(dbgInfo);
         }
     }
 
-    private ProfileEntryValueDto makeDateProfileEntryValueDto(final DataObject dataObject)
-            throws ProtocolAdapterException {
-        final CosemDateTimeDto cosemDateTime = this.dlmsHelper.convertDataObjectToDateTime(dataObject);
-        return new ProfileEntryValueDto(cosemDateTime.asDateTime().toDate());
+    private ProfileEntryValueDto makeDateProfileEntryValueDto(final DataObject dataObject,
+            ProfileEntryDto previousProfileEntryDto, int timeInterval) {
+        final CosemDateTimeDto cosemDateTime;
+        try {
+            cosemDateTime = this.dlmsHelper.convertDataObjectToDateTime(dataObject);
+            return new ProfileEntryValueDto(cosemDateTime.asDateTime().toDate());
+        } catch (ProtocolAdapterException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)